src/models/tests/goals.test.js

```diff
@@ -112,145 +115,154 @@ describe('Goals', () => {
+    expect(instance.lastCeasedSuspendedAt).toEqual(undefined);
     expect(instance.lastClosedAt).toEqual(undefined);
 
     instance = {
       status: '',
       changed: () => ['status'],
     };
     instance.set = (name, value) => { instance[name] = value; };
-    autoPopulateStatusChangeDates(null, instance);
+    options = { fields: [] };
+    autoPopulateStatusChangeDates(null, instance, options);
     expect(instance.firstNotStartedAt).toEqual(undefined);
     expect(instance.firstInProgressAt).toEqual(undefined);
-    expect(instance.firstSuspendedAt).toEqual(undefined);
+    expect(instance.firstCeasedSuspendedAt).toEqual(undefined);
     expect(instance.firstClosedAt).toEqual(undefined);
     expect(instance.lastNotStartedAt).toEqual(undefined);
     expect(instance.lastInProgressAt).toEqual(undefined);
-    expect(instance.lastSuspendedAt).toEqual(undefined);
+    expect(instance.lastCeasedSuspendedAt).toEqual(undefined);
     expect(instance.lastClosedAt).toEqual(undefined);
 
     instance = {
       status: GOAL_STATUS.DRAFT,
       changed: () => ['status'],
     };
     instance.set = (name, value) => { instance[name] = value; };
-    autoPopulateStatusChangeDates(null, instance);
+    options = { fields: [] };
+    autoPopulateStatusChangeDates(null, instance, options);
     expect(instance.firstNotStartedAt).toEqual(undefined);
     expect(instance.firstInProgressAt).toEqual(undefined);
-    expect(instance.firstSuspendedAt).toEqual(undefined);
+    expect(instance.firstCeasedSuspendedAt).toEqual(undefined);
     expect(instance.firstClosedAt).toEqual(undefined);
     expect(instance.lastNotStartedAt).toEqual(undefined);
     expect(instance.lastInProgressAt).toEqual(undefined);
-    expect(instance.lastSuspendedAt).toEqual(undefined);
+    expect(instance.lastCeasedSuspendedAt).toEqual(undefined);
     expect(instance.lastClosedAt).toEqual(undefined);
 
     instance = {
       status: GOAL_STATUS.NOT_STARTED,
       changed: () => ['status'],
     };
     instance.set = (name, value) => { instance[name] = value; };
+    options = { fields: [] };
     const ts = (new Date()).getTime();
-    autoPopulateStatusChangeDates(null, instance);
+    autoPopulateStatusChangeDates(null, instance, options);
     expect((Date.parse(instance.firstNotStartedAt)))
       .toBeLessThan(ts);
     expect(instance.firstInProgressAt).toEqual(undefined);
-    expect(instance.firstSuspendedAt).toEqual(undefined);
+    expect(instance.firstCeasedSuspendedAt).toEqual(undefined);
     expect(instance.firstClosedAt).toEqual(undefined);
     expect((Date.parse(instance.lastNotStartedAt)))
       .toBeLessThan(ts);
     expect(instance.lastInProgressAt).toEqual(undefined);
-    expect(instance.lastSuspendedAt).toEqual(undefined);
+    expect(instance.lastCeasedCeasedSuspendedAt).toEqual(undefined);
     expect(instance.lastClosedAt).toEqual(undefined);
     expect(instance.lastNotStartedAt).toEqual(instance.firstNotStartedAt);
 
     await sleep(1000);
```

Comment:
You don't need an await, the function you wrote doesn't return a promise so I don't think its doing anything

Also, do we want to test the possibility that fields is not an array? I notice you have an if statement setting it as one in the above code